src/main.rs

```diff
@@ -31,16 +32,30 @@ use structopt::StructOpt;
 #[tokio::main]
 async fn main() -> ResultWithDefaultError<()> {
     let parsed_args = CommandLineArguments::from_args();
-    return execute_subcommand(parsed_args.cmd).await;
+    match execute_subcommand(parsed_args.cmd).await {
+        Ok(()) => Ok(()),
+        Err(error) => {
+            // We are catching the error and pretty printing it instead of letting the
+            // program error. Since we are not meant to be used other programs, I think
+            // it's fine to always return a 0 error code, but we might wanna revisit this.
+            print!("{}", error);
```

Comment:
Agree with outputting the error message, but I think it would be nicer to be compliant with unix standards and have error exit codes i would still like the ability for the user to do some crazy pipelines, if they want to.
but this must be done in a follow up, let's merge this one for now.